### PR TITLE
Migrated to biomejs, reverted url delete, removed eslint comment.

### LINKED ISSUE #271: Upgrade to ESLint 9.x
Let's clean up the extraneous eslint plugins and configs we have. The simpler, the better ➖

### PR DESCRIPTION
**Pull request by KemingHe**
**Reviewer Requested**: @wei 

**Timeline**:
```
#271 [Issue] Upgrade to ESLint 9.x, in-progress, open
|
├── #286 [PR] Migration to BiomeJS, unsuccessful, closed
|
└── #290 [PR] Migration polish, in-progress, open

#272 [Issue] Upgrade to NextJS 14.x, in-progress, open
|
├── #273 [PR] Simple package.json edit, unsuccessful, open
|
└── #291 [Issue] Migration to alias import, in-progress, open
    |
    └── #290 [PR] Migration polish, in-progress, open
        |
        └── #??? [PR] Migration to NextJS 14.x w/ app router, planning, not-open
```

**Update**: Latest #290 fix will address:
- [x] `/* eslint-ignore */` equivalent in BiomeJS
- [x] Inline ignore, file ignore, path ignore, pattern ignore
- [x] Migrate from relative path to aliasing, see #291 
- [x] Revert url changes to remove `undefined` with `delete`
- [x] Revert `noDelete` flag to true
- [x] List specific rules migrated in a neatly organized table in `biome.json.md` file
- [ ] Make sure all tests passes, all behaviors stay the same as current production

*This commit will be signed off by @KemingHe*
> Comment update: more context

### CODE REVIEW

**Comment by KemingHe** [comment]
@wei By the way, would you consider a staging branch before production? The current setup is auto-deploy on `master`, which puts a lot of pressure on the reviewer, speaking from experience.

File: src/components/configuration/config.tsx [hunk 1]
```diff
@@ -30,9 +31,11 @@ const Config = ({ repository }: ConfigProp) => {
   const handleChanges = (changes: { value: any; key: keyof ConfigType }[]) => {
     let newConfig: ConfigType = { ...config }
     const urlParams = router.query
+
     // Remove extraneous params from route
-    delete urlParams._owner
-    delete urlParams._name
+    urlParams._owner = undefined
+    urlParams._name = undefined
```
Comment (wei) [inline src/components/configuration/config.tsx:37]: Still seeing this hmm, please undo.

File: biome.json [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+    "lineEnding": "lf",
+    "lineWidth": 80,
+    "attributePosition": "auto"
+  },
+  "linter": {
+    "enabled": true,
+    "rules": {
+      "recommended": true,
+      "performance": {
+        "noDelete": "warn"
```
Comment (wei) [inline biome.json:39]: This is fine, just curious does biomejs have an inline ignore or do you have to set it globally in the config?

**Review by wei — CHANGES_REQUESTED** [review]
Overall LGTM, please review comments.

In addition, Chinese characters are no longer being rendered correctly.

Production: [](https://socialify.git.ci/wei/socialify/image?description=1&descriptionEditable=%F0%9F%A4%96%20Chinese%EF%BC%9A%E4%B8%AD%E6%96%87%20%3C%3C%3C%20should%20display%20correctly&language=1&name=1&owner=1&stargazers=1&theme=Light)

This PR preview:
[](https://socialify-git-fork-keminghe-master-github-socialify.vercel.app/wei/socialify/image?description=1&descriptionEditable=%F0%9F%A4%96%20Chinese%EF%BC%9A%E4%B8%AD%E6%96%87%20%3C%3C%3C%20should%20display%20correctly&language=1&name=1&owner=1&stargazers=1&theme=Light)

I suspect `common/font.ts` changes is causing Chinese characters to have render issues. Could you take a look?

Since that file is pulled directly from an upstream source, can we ignore that file completely using biomejs, similar to eslint-ignore?

**Comment by wei** [comment]
> @wei By the way, would you consider a staging branch before production? The current setup is auto-deploy on `master`, which puts a lot of pressure on the reviewer, speaking from experience.

Thanks for the suggestion. We are using Vercel for Pull Request previews which can act as staging. This is how I am able to test your changes live: https://socialify-git-fork-keminghe-master-github-socialify.vercel.app/

File: src/components/configuration/config.tsx (hunk 1 above)
Comment (KemingHe) [reply src/components/configuration/config.tsx:37]: My bad, I think I need to tweak my BiomeJS setup a bit to not have it automatically change it back after my manual reversion. I'll keep you posted.

File: biome.json (hunk 2 above)
Comment (KemingHe) [reply biome.json:39]: I will further investigate the fine-grained control option and include it in the next request. Thank you for pointing this out.

**Comment by wei** [comment]
@KemingHe no need to keep closing prs. you could just add commits to your branch which will update the pr automatically 😁

**Comment by KemingHe** [comment]
Back to working on more polishing...

**Comment by KemingHe** [comment]
> Overall LGTM, please review comments.
> 
> In addition, Chinese characters are no longer being rendered correctly.
> 
> Production: [](https://socialify.git.ci/wei/socialify/image?description=1&descriptionEditable=%F0%9F%A4%96%20Chinese%EF%BC%9A%E4%B8%AD%E6%96%87%20%3C%3C%3C%20should%20display%20correctly&language=1&name=1&owner=1&stargazers=1&theme=Light)
> 
> This PR preview: [](https://socialify-git-fork-keminghe-master-github-socialify.vercel.app/wei/socialify/image?description=1&descriptionEditable=%F0%9F%A4%96%20Chinese%EF%BC%9A%E4%B8%AD%E6%96%87%20%3C%3C%3C%20should%20display%20correctly&language=1&name=1&owner=1&stargazers=1&theme=Light)
> 
> I suspect `common/font.ts` changes is causing Chinese characters to have render issues. Could you take a look?
> 
> Since that file is pulled directly from an upstream source, can we ignore that file completely using biomejs, similar to eslint-ignore?

Hi @wei, I'm having difficulties recreating the language rendering bug you encountered, this is my result with the same preview, everything looks okay so far:

Screenshot 2024-05-05 at 11 53 32 AM

Screenshot 2024-05-05 at 11 58 54 AM

URL: https://socialify-git-fork-keminghe-master-github-socialify.vercel.app/wei/socialify/image?_name=undefined&_owner=undefined&description=1&descriptionEditable=%F0%9F%A4%96%20Chinese%3A%20%E4%B8%AD%E6%96%87%20%20%3C%3C%3C%20should%20display%20correctly&language=1&name=1&owner=1&stargazers=1&theme=Light

Could you please provide more details on what could be the cause on your end? Much appreciated!

**Comment by KemingHe** [comment]
> @wei @KemingHe Oh wait, I see, the preview works but the actual image rendering doesn't. Hmmm... Give me a sec.

And I also broke the download dropdown somehow, gotta fix that quick as well 😅

**Comment by KemingHe** [comment]
@wei I should probably add `playwright` or `puppeteer` behavior testing to this, these usage bugs are so hard to find. 🤔

**Comment by KemingHe** [comment]
Okay, after some initial testing, it's not the url issue, but some internal issue, will keep working on it.

**Comment by wei** [comment]
> @wei I should probably add `playwright` or `puppeteer` behavior testing to this, these usage bugs are so hard to find. 🤔

Good idea, we should actually prioritize adding tests like the upstream repo using [__image_snapshots__](https://github.com/vercel/satori/tree/main/test/__image_snapshots__) to ensure the rendered image match snapshots with different parameters, Chinese characters, logos, emojis, etc. Afterwards an e2e test of the frontend will be nice too.

**Comment by wei** [comment]
Ideally the migration to biomejs should only be coding style change and not change any functionality.

**Comment by KemingHe** [comment]
I'm archiving my current migration and starting from scratch. Closing pull request since I'm changing fork.

**Pull request closed without merging**